src/Core/Style.js

```diff
@@ -919,17 +958,16 @@ class Style {
      * @param {Boolean} canBeFilled - true if feature.type == FEATURE_TYPES.POLYGON.
      */
     applyToCanvasPolygon(txtrCtx, polygon, invCtxScale, canBeFilled) {
-        const context = this.context;
         // draw line or edge of polygon
         if (this.stroke) {
             // TO DO add possibility of using a pattern (https://github.com/iTowns/itowns/issues/2210)
-            this._applyStrokeToPolygon(txtrCtx, invCtxScale, polygon, context);
+            this._applyStrokeToPolygon(txtrCtx, invCtxScale, polygon);
         }
 
         // fill inside of polygon
-        if (canBeFilled && this.fill) {
+        if (canBeFilled && (this.fill.pattern || this.fill.color)) {
             // canBeFilled can be move to StyleContext in the later PR
```

Comment:
Could we remove canBeFilled ? Its value is feature.type == FEATURE_TYPES.POLYGON . We are in a method named `applyToCanvasPolygon` so aren't already expecting to handle only polygons ?

Reply:
In fact `applyToCanvasPolygon` is called in the function `drawPolygon()` in Feature2Texture.js and that function is called on any feature.type that isn't a Point. Thus is include polygon and line, the `canBeFilled` is then the way to distinguish both.
It's the name that is maybe to rethink at some point..

Reply:
Ah ok thanks, we should rename applyToCanvasPolygon then, I agree. Let's do it in another PR